tck/features/ListOperations.feature

```diff
@@ -0,0 +1,748 @@
+      """
+    Then the result should be:
+      | foo                  |
+      | [false, true, false] |
+    And no side effects
+
+  Scenario: Execute n[0]
+    When executing query:
+      """
+      RETURN [1, 2, 3][0] AS value
+      """
+    Then the result should be:
+      | value |
+      | 1     |
+    And no side effects
+
+  Scenario: Extract eagerly
+    And having executed:
+      """
+      CREATE (:Label1 {name: 'original'})
+      """
+    When executing query:
+      """
+      MATCH (a:Label1)
+      WITH collect(a) AS nodes
+      WITH nodes, extract(x IN nodes | x.name) AS oldNames
+      UNWIND nodes AS n
+      SET n.name = 'newName'
+      RETURN n.name, oldNames
+      """
+    Then the result should be:
+      | n.name    | oldNames   |
+      | 'newName' | ['original'] |
+    And the side effects should be:
+      | +properties | 1 |
+      | -properties | 1 |
+
+  Scenario: Filter eagerly
+    And having executed:
+      """
+      CREATE (:Label1 {name: 'original'})
+      """
+    When executing query:
+      """
+      MATCH (a:Label1)
+      WITH collect(a) AS nodes
+      WITH nodes, filter(x IN nodes WHERE x.name = 'original') AS noopFiltered
+      UNWIND nodes AS n
+      SET n.name = 'newName'
+      RETURN n.name, length(noopFiltered)
+      """
+    Then the result should be:
+      | n.name    | length(noopFiltered)   |
+      | 'newName' | 1                      |
+    And the side effects should be:
+      | +properties | 1 |
+      | -properties | 1 |
+
+  Scenario: Length on filter
+    When executing query:
+      """
+      MATCH (n)
+      OPTIONAL MATCH (n)-[r]->(m)
+      RETURN length(filter(x IN collect(r) WHERE x <> null)) AS cn
+      """
+    Then the result should be:
+      | cn |
+      | 0  |
+    And no side effects
+
+  # List lookup based on parameters
+
+  Scenario: Use list lookup based on parameters when there is no type information
+    And parameters are:
+      | expr | ['Apa'] |
+      | idx  | 0       |
+    When executing query:
+      """
+      WITH $expr AS expr, $idx AS idx
+      RETURN expr[idx] AS value
+      """
+    Then the result should be:
+      | value |
+      | 'Apa' |
+    And no side effects
+
+  Scenario: Use list lookup based on parameters when there is lhs type information
+    And parameters are:
+      | idx | 0 |
+    When executing query:
+      """
+      WITH ['Apa'] AS expr
+      RETURN expr[$idx] AS value
+      """
+    Then the result should be:
+      | value |
+      | 'Apa' |
+    And no side effects
+
+  Scenario: Use list lookup based on parameters when there is rhs type information
+    And parameters are:
+      | expr | ['Apa'] |
+      | idx  | 0       |
+    When executing query:
+      """
+      WITH $expr AS expr, $idx AS idx
+      RETURN expr[toInteger(idx)] AS value
+      """
+    Then the result should be:
+      | value |
+      | 'Apa' |
+    And no side effects
+
+  # Failures at runtime
```

Comment:
Perhaps we could partition the negative tests into its own feature? Perhaps `ListOperationsNegative.feature` or `ListOperationsErrors.feature`?

Reply:
I think they belong together and 3 scenarios also don't justify their own feature file. But I'll leave the last call to you @Mats-SX ;)